### PR TITLE
Skip service port for scraper receivers

### PR DESCRIPTION
**Pull request by srikanthccv**
A good chunk of receivers in the collector are scraper kind for which we don't need to create a service port.

### CODE REVIEW

File: pkg/collector/parser/receiver.go [hunk 1]
```diff
@@ -77,8 +77,49 @@ func IsRegistered(name string) bool {
 var (
 	endpointKey      = "endpoint"
 	listenAddressKey = "listen_address"
+	scraperReceivers = []string{
```
Comment (TylerHelmuth) [inline pkg/collector/parser/receiver.go:80]: Let's make `scraperReceivers` a `map[string]bool` so the lookup is is more efficient.

File: pkg/collector/parser/receiver.go [hunk 2]
```diff
@@ -77,8 +77,45 @@ func IsRegistered(name string) bool {
 var (
 	endpointKey      = "endpoint"
 	listenAddressKey = "listen_address"
+	scraperReceivers = map[string]struct{}{
```
Comment (jaronoff97) [inline pkg/collector/parser/receiver.go:80]: Should we not include `prometheus` in this list too?

Comment (srikanthccv) [reply pkg/collector/parser/receiver.go:80]: There is no top-level `endpoint` for Prometheus config. However, it is covered with a note just below the lines of this change. I don't mind moving it to this list.

Comment (jaronoff97) [reply pkg/collector/parser/receiver.go:80]: yeah i think we should move it to this list given it provides the same functionality as the case in the switch below.

Comment (srikanthccv) [reply pkg/collector/parser/receiver.go:80]: Updated

**Comment by jaronoff97** [comment]
@srikanthccv looks like there's a linting failure

**Comment by srikanthccv** [comment]
It should be fixed now.

**Review by TylerHelmuth — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**